For...of with HTMLCollection

Hi,

Getting a compile error when trying to compile:

```
// ul is an HTMLElement
for (let child of ul.children) {
    //body
}
```

I get the following error: `error TS2495: Type 'HTMLCollection' is not an array type or a string type.`

Is there a way to get type support for this? Or is too much work to bother with?

--- a/src/lib/dom.iterable.d.ts
+++ b/src/lib/dom.iterable.d.ts
@@ -87,6 +87,14 @@ interface NodeListOf<TNode extends Node> {
     [Symbol.iterator](): IterableIterator<TNode>;
 }
 
+interface HTMLCollectionBase {
+    [Symbol.iterator](): IterableIterator<Element>;
+}
+
+interface HTMLCollectionOf<T extends Element> {
+    [Symbol.iterator](): IterableIterator<T>;
+}
+
 interface URLSearchParams {
     /**
      * Returns an array of key, value pairs for every entry in the search params

--- a/src/lib/dom.iterable.d.ts
+++ b/src/lib/dom.iterable.d.ts
@@ -4,23 +4,6 @@ interface DOMTokenList {
     [Symbol.iterator](): IterableIterator<string>;
 }
 
-interface FormData {
-    /**
-     * Returns an array of key, value pairs for every entry in the list
-     */
-    entries(): IterableIterator<[string, string | File]>;
-    /**
-     * Returns a list of keys in the list
-     */
-    keys(): IterableIterator<string>;
-    /**
-     * Returns a list of values in the list
-     */
-    values(): IterableIterator<string | File>;
-
-    [Symbol.iterator](): IterableIterator<string | File>;
-}
-
 interface Headers {
     [Symbol.iterator](): IterableIterator<[string, string]>;
     /**
@@ -93,6 +76,23 @@ interface HTMLCollectionBase {
 
 interface HTMLCollectionOf<T extends Element> {
     [Symbol.iterator](): IterableIterator<T>;
+}
+
+interface FormData {
+    /**
+     * Returns an array of key, value pairs for every entry in the list
+     */
+    entries(): IterableIterator<[string, string | File]>;
+    /**
+     * Returns a list of keys in the list
+     */
+    keys(): IterableIterator<string>;
+    /**
+     * Returns a list of values in the list
+     */
+    values(): IterableIterator<string | File>;
+
+    [Symbol.iterator](): IterableIterator<string | File>;
 }
 
 interface URLSearchParams {

